### DISCUSSION TITLE
Mjpeg server using uWebSocket?

### DISCUSSION THREAD

**Discussion by badpaybad** [uNetworking/uWebSockets#1362] [Q&A]
I want to read my camera and public it.
I use opencv and uSW::App .
- One thread to capture frame , put into queue
- an routing handle using uSW::App to loop infinity to dequeue then send to client

So I tried this way. It worked. But I got blocked other url  eg:  _app->get("/about ", ...) { }

Could someone can share the way to do?

snip code: 

` 
				uWS::App *_app;

				  _app->get("/video",
				  [qBuf, lockGlobal](auto *res, auto *req)
				  {
					  res->writeHeader("Content-Type", "multipart/x-mixed-replace; boundary=frame");
					  std::vector<uchar> buff_bgr;
					  while (1)
					  {
						  lockGlobal->try_lock();
						  if (qBuf->size() > 0)
						  {
							  buff_bgr = qBuf->front();
							  qBuf->pop();
						  }
						  lockGlobal->unlock();
						  auto imgc = "--frame\r\nContent-Type: image/jpeg\r\nContent-Length: " +

									  std::to_string(buff_bgr.size()) + "\r\n\r\n" +
									  std::string(buff_bgr.begin(), buff_bgr.end()) + "\r\n";
						  res->write(imgc);
						  std::this_thread::sleep_for(std::chrono::milliseconds(30));
					  }
				  });
`

**Comment by Lssikkes** [comment]
You can't use sleep_for, in this case you could set up a timer that runs every 30 milliseconds instead of the while loop. You need some way to handle connection aborts as well. (use us_create_timer, us_timer_set, us_timer_ext to pass through arguments, and us_timer_close on res->onAborted). Ideally you'd also handle res->write's return value (if it returns false, stop sending more data, otherwise you're creating a ton of backpressure)

**Reply by Lssikkes** [reply]
I'd also recommend to use buff_bgr.swap(qBuf->front()); instead of buff_bgr = qBuf->front(); - should save a copy

**Comment by badpaybad** [comment]
I tried to figure out, but I can not improve my code. Could you spend time to check please?
https://github.com/badpaybad/rplidaropencv/blob/main/libs/MjpegHttpServer.cpp

I am mostly working in C#, recently I need to work with C/C++
